lib/record_store/zone.rb

```diff
@@ -56,22 +56,49 @@ def yaml_files
     validate :validate_provider_can_handle_zone_records
 
     def self.from_yaml_definition(name, definition)
+      definition['records'] ||= []
+      Dir["#{RecordStore.zones_path}/#{name}/*__*.yml"].each do |record_file|
```

Comment:
I think this logic should live in the [`load_yaml_file` private method above](https://github.com/Shopify/record_store/blob/30247c9f706441fc454b9ecc61e957c225ea881b/lib/record_store/zone.rb#L29-L35). We can rename it to something more general.